android/src/main/java/com/zoontek/rnpermissions/RNPermissionsModule.java

```diff
@@ -141,16 +143,13 @@ private boolean permissionExists(final String permission) {
 
   @ReactMethod
   public void checkNotifications(final Promise promise) {
-    final boolean enabled = NotificationManagerCompat
-      .from(getReactApplicationContext()).areNotificationsEnabled();
-
-    final WritableMap output = Arguments.createMap();
-    final WritableMap settings = Arguments.createMap();
-
-    output.putString("status", enabled ? GRANTED : BLOCKED);
-    output.putMap("settings", settings);
+    if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+      this.checkNotificationsCompat(promise);
+      return;
+    }
 
-    promise.resolve(output);
+    String fieldName = this.getFieldName("android.permission.POST_NOTIFICATIONS");
```

Comment:
This make `checkPermission` to be called with `"POST_NOTIFICATIONS"` instead of `"android.permission.POST_NOTIFICATIONS"`, which will always result on `"unavailable"`